core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -193,24 +147,24 @@ public void dispatchRequest(final RestRequest request, final RestChannel channel
                     threadContext.putHeader(key, httpHeader);
                 }
             }
-            if (filters.length == 0) {
-                executeHandler(request, channel, client);
+
+            final RestHandler rawHandler = getHandler(request);
+            final RestHandler handler = handlerWrapper.apply(rawHandler);
+
+            if (handler != null) {
```

Comment:
I have null meaning "don't use a wrapper", in which case we convert the null here to an identity wrapper to simplify the code which uses the wrapper.